presidio-analyzer/presidio_analyzer/recognizer_registry/recognizer_registry.py

```diff
@@ -276,20 +283,43 @@ def add_recognizer(self, recognizer: EntityRecognizer) -> None:
 
         self.recognizers.append(recognizer)
 
-    def remove_recognizer(self, recognizer_name: str) -> None:
+    def remove_recognizer(
+        self, recognizer_name: str, language: Optional[str] = None
+    ) -> None:
         """
         Remove a recognizer based on its name.
 
         :param recognizer_name: Name of recognizer to remove
+        :param language: The supported language of the recognizer to be removed,
+        in case multiple recognizers with the same name are present,
+        and only one should be removed.
         """
-        new_recognizers = [
-            rec for rec in self.recognizers if rec.name != recognizer_name
-        ]
-        logger.info(
-            "Removed %s recognizers which had the name %s",
-            str(len(self.recognizers) - len(new_recognizers)),
-            recognizer_name,
-        )
+
+        if not language:
```

Comment:
Now that we create multiple instances of a recognizer for each language, one might need to remove just one instance (say the german CreditCardRecognizer) and not all credit card recognizers